dvc/repo/brancher.py

```diff
@@ -77,7 +77,7 @@ def brancher(  # noqa: E302
         from dvc.fs import GitFileSystem
 
         for sha, names in found_revs.items():
-            self.__dict__.pop("index", None)
+            self._reset_cached_indecies()  # pylint: disable=W0212
```

Comment:
This is not actually required as the `self.fs` setter calls `Repo._reset()`.
See https://github.com/iterative/dvc/pull/8407#discussion_r992043019.

Anyway, the changes look good to me.